src/Sylius/Component/Core/Model/ProductVariant.php

```diff
@@ -345,6 +345,19 @@ public function getImages()
     /**
      * {@inheritdoc}
      */
+    public function getImage()
+    {
+        if ($this->images->isEmpty()) {
+            $image = $this->getProduct()->getImage();
+        } else {
+            $image = $this->images->first();
```

Comment:
I think you can just return here: ``return $this->images->first();``. It will return null in case there is no image.